### DISCUSSION TITLE
Successful workflow not archived for a long time?

### DISCUSSION THREAD

**Discussion by zhucan** [argoproj/argo-workflows#13150] [Q&A]
Argo workflow version is: v3.4.9

1. Get the "succeeded" argo workflow,  the results is:
```
kubectl get workflow -n argo-data-closeloop | grep  prod--prod--perception-autolabel-lidar-ground-reconstructizvtq6
prod--prod--perception-autolabel-lidar-ground-reconstructizvtq6   Succeeded   5h57m
```

2. The ttlStrategy is:
```yaml
  ttlStrategy:
    secondsAfterCompletion: 300
```
it means the succeeded argo workflow will be archived after 300s ago?

3. The labels of the argo workflow is:
```yaml
  labels:
    workflows.argoproj.io/completed: "true"
    workflows.argoproj.io/creator: system-serviceaccount-argo-data-closeloop-argo-data-closeloop
    workflows.argoproj.io/phase: Succeeded
    workflows.argoproj.io/workflow-archiving-status: Pending
```
There are so many "succeeded" argo workflows not  archived more the  5m, I don't know why?
```
prod--prod--perception-autolabel-lidar-ground-reconstructizbw6w   Succeeded   5h28m   
prod--prod--perception-autolabel-lidar-ground-reconstructizrrhs   Succeeded   5h18m   
prod--prod--perception-autolabel-lidar-ground-reconstructizt56t   Succeeded   5h46m   
prod--prod--perception-autolabel-lidar-ground-reconstructiztm98   Succeeded   5h53m   
prod--prod--perception-autolabel-lidar-ground-reconstructizvtq6   Succeeded   5h45m   
prod--prod--perception-autolabel-occupancy-reconstruction-5h947   Succeeded   170m    
prod--prod--perception-autolabel-occupancy-reconstruction-5hrjd   Succeeded   154m    
prod--prod--perception-autolabel-occupancy-reconstruction-85h7n   Succeeded   3h10m   
prod--prod--perception-autolabel-pcd-compensation-1-1-33--bhn5h   Succeeded   12m     
prod--prod--perception-autolabel-three-d-semantic-1-1-41--5hb7d   Succeeded   6h37m   
prod--prod--perception-autolabel-three-d-semantic-1-1-41--8f5h8   Succeeded   6h43m   
prod--prod--perception-autolabel-three-d-semantic-1-1-41--95h44   Succeeded   6h49m   
prod--prod--perception-autolabel-three-d-semantic-1-1-41--k5h6v   Succeeded   6h43m   
prod--prod--perception-autolabel-three-d-semantic-1-1-41--k65hm   Succeeded   6h48m   
prod--prod--perception-autolabel-three-d-semantic-1-1-41--tpw5h   Succeeded   6h42m   
prod--prod--perception-autolabel-three-d-semantic-1-1-41--vb5hl   Succeeded   6h41m   
prod--prod--perception-autolabel-three-d-semantic-1-1-41--w5hq4   Succeeded   6h50m   
prod--prod--perception-autolabel-three-d-semantic-1-1-41--xc45h   Succeeded   6h45m   
prod--prod--perception-autolabel-track-refine-1-1-17--42casv5bn   Succeeded   5h10m   
prod--prod--perception-autolabel-track-refine-1-1-17--51b9n9hvg   Succeeded   5h41m   
prod--prod--perception-autolabel-track-refine-1-1-17--66489fjv6   Succeeded   5h28m   
prod--prod--perception-autolabel-track-refine-1-1-17--becdkh5kg   Succeeded   5h17m   
prod--prod--perception-autolabel-track-refine-1-1-17--f4e7zhtb6   Succeeded   5h23m   
```

**Comment by zhucan** [comment]
other question is,  is there way to manual archiving the argo workflow if the status is "succeeded"?

**Reply by agilgur5** [reply]
No there is not, the Controller does this automatically. See my [below answer](https://github.com/argoproj/argo-workflows/discussions/13150#discussioncomment-9692904) for more details

**Comment by agilgur5** [comment]
> Argo workflow version is: v3.4.9

This is an older patch, I would recommend upgrading to a newer patch, such as 3.4.16.

> it means the succeeded argo workflow will be archived after 300s ago?

Not quite. The Workflow Archive does not require a TTL to be set. Workflows are labeled for archiving [when they are completed](https://github.com/argoproj/argo-workflows/blob/88304568af5b8222344cde8e5b96d1a8484a50f8/workflow/controller/operator.go#L2362). This label is then [picked up by the Informer](https://github.com/argoproj/argo-workflows/blob/88304568af5b8222344cde8e5b96d1a8484a50f8/workflow/controller/controller.go#L1070-L1084) which kicks off archiving.

A `ttlStrategy` will delete a Workflow from the cluster, independent of whether the Workflow is archived or not. Workflows in the archive can also be automatically deleted via [`archiveTTL`](https://argo-workflows.readthedocs.io/en/latest/workflow-archive/#archive-ttl).

> There are so many "succeeded" argo workflows not archived more the 5m, I don't know why?

I would guess this is because some Workflow processing is stuck. For instance, per your labels:

> ```yaml
>     workflows.argoproj.io/workflow-archiving-status: Pending
> ```

The archiving never finished, so the Controller is not deleting the in-cluster Workflow resource until it is archived, i.e. until it is safe to do so without losing history.

Your logs may reveal why it has not finished archiving, likely due to a misconfiguration somewhere.

**Comment by zhucan** [comment]
@agilgur5 Thanks for your responses, I haven't no plan to upgrade the argo workfow.

One argo workflow can not be archived for a long time,  the logs of controller is:
![image](https://github.com/argoproj/argo-workflows/assets/13997135/fc7571c8-b1c2-408f-ba5f-48d2769be1ea)

```
kubectl get workflow --all-namespaces  | grep prod--prod--perception-sync-data-frame-downloader-1-1-35--glmms
argo-data-closeloop   prod--prod--perception-sync-data-frame-downloader-1-1-35--glmms   Succeeded   13h 
```

**Reply by agilgur5** [reply]
>  I haven't no plan to upgrade the argo workfow.

Er, ok then, I mean you might have bugs and would have some insecure deps if you're not on the latest patch 🤷  Heck there might even have been a bugfix for archiving; I don't know, you're 7+ patches behind.

> the logs of controller is:

Please use text code snippets for logs and not images; text is substantially [more accessible](https://meta.stackoverflow.com/a/285557/3431180) than images

> One argo workflow can not be archived for a long time

Is it just one or all of them? If it's just one then I would guess there is something incomplete with the Workflow.
I can't really say anything meaningful with that little information and no reproduction.

I'm not sure if you have more logs, but those logs only mention pending archiving and there is no mention of archiving starting, which is where I would anticipate some error message if the Workflow is completed or if other Workflows succeed.

**Reply by zhucan** [reply]
> Please use text code snippets for logs and not images; text is substantially [more accessible](https://meta.stackoverflow.com/a/285557/3431180) than images

Because of the controller oom, and the logs had been archived to the es

> Is it just one or all of them? If it's just one then I would guess there is something incomplete with the Workflow.
I can't really say anything meaningful with that little information and no reproduction.

yes, I think it's slowly to archive,  not stuck

**Reply by agilgur5** [reply]
> Because of the controller oom

That sounds like an important detail you didn't include previously...

> and the logs had been archived to the es

I don't know what "es" means... not sure if you meant S3?

In any case, you can still copy your logs without a screenshot. Please follow common best practices when asking for support, as I [already linked](https://meta.stackoverflow.com/a/285557/3431180) (note that has 1k+ upvotes) and do not ignore them. See also [this link](https://idownvotedbecau.se/imageofanexception/)

> yes, I think it's slowly to archive, not stuck

Ok, then sounds like this can be closed? If your Controller is starved, as an OOM would suggest, that's certainly possible as it would cause a bottleneck or crash; processes are not expected to run starved.

Also I didn't ask a "yes/no" question, so you did not quite answer the question

**Comment by zhucan** [comment]
> That sounds like an important detail you didn't include previously...

Our k8s cluster create more than 18000+ workflows, but the cpu and mem limit is 32c and 256G,  but the controller use the mem more than 256G，so the container was killed by kernel

>  I don't know what "es" means... not sure if you meant S3?
 
es means elasticsearch, becuase of the oom, so the logs of the contained will be cleaned,  the old logs was archived to the elasticsearch.
